releasenotes/notes/add-mcx-plugins-85e5b248692a36db.yaml

```diff
@@ -0,0 +1,38 @@
+---
+features_synthesis:
+  - |
+    Added synthesis functions :func:`.synth_mcx_gray_code` and :func:`.synth_mcx_v24`
+    that synthesize multi-controlled X gates. These functions do not require additional
+    ancilla qubits.
+  - |
+    Added synthesis functions :func:`.synth_c3x` and  :func:`.synth_c4x` that
+    synthesize 3-controlled and 4-controlled X-gates respectively.
+features_transpiler:
+  - |
+    Added multiple high-level-synthesis plugins for synthesizing an :class:`.MCXGate`:
+
+    * :class:`.MCXSynthesisNCleanM15`, based on :func:`.synth_mcx_n_clean_m15`.
+    * :class:`.MCXSynthesisNDirtyI15`, based on :func:`.synth_mcx_n_dirty_i15`.
+    * :class:`.MCXSynthesis1CleanB95`, based on :func:`.synth_mcx_1_clean_b95`.
+    * :class:`.MCXSynthesisV24`, based on :func:`.synth_mcx_v24`.
+    * :class:`.MCXSynthesisGrayCode`, based on :func:`.synth_mcx_gray_code`.
+
+    As well:
+
+    * :class:`.MCXSynthesisGrayDefault`, choosing the most efficient synthesis
+      method based on the number of clean and dirty ancilla qubits available.
+
+deprecations:
```

Comment:
We don't mention pending deprecation afaik, I think this can be removed 🙂

Reply:
done in [5a8ba64](https://github.com/Qiskit/qiskit/pull/12961/commits/5a8ba642a7024566ef8d78d06a2dd49526d66662)